crates/ide/src/references.rs

```diff
@@ -112,6 +114,26 @@ pub(crate) fn find_all_refs(
     }
 }
 
+fn retain_import_usages(usages: &mut UsageSearchResult, sema: &Semantics<'_, RootDatabase>) {
+    for (file_id, refs) in &mut usages.references {
+        refs.retain(|x| {
+            let file_sema = sema.parse(file_id.clone()).syntax().clone();
+
+            let maybe_node = file_sema.child_or_token_at_range(x.range.clone());
+
+            if let Some(node) = maybe_node {
+                let res = match node {
+                    syntax::NodeOrToken::Node(x) => !matches!(x.kind(), USE),
+                    syntax::NodeOrToken::Token(_) => true,
+                };
+                res
+            } else {
+                true
+            }
+        });
```

Comment:
This still uses the range instead of the node of `FileReference`,
```suggestion
        refs.retain(|it| {
            match if.name.as_name_ref() {
                Some(name_ref) => name_ref.syntax().ancestors().any(|it| !matches(it.kind(), USE)),
                None => true,
            }
        });
```

Reply:
Sadly i'm too way bad with rust-analyzer's query system. I fixed and tried your suggestion, but looks like `ancestors()` dont work for this case. 

dbg shows only something like:

NAME_REF@5547..5554
  IDENT@5547..5554 "MainApp"

Reply:
Looks like I find the problem. we need inverse-`any()`